sdcm/nemesis.py

```diff
@@ -2090,10 +2090,15 @@ def call_next_nemesis(self):
         self.execute_disrupt_method(disrupt_method=next(self.disruptions_cycle))
 
     @latency_calculator_decorator(legend="Run repair process with nodetool repair")
-    def repair_nodetool_repair(self, node=None, publish_event=True):
-        node = node if node else self.target_node
-        with adaptive_timeout(Operations.REPAIR, node, timeout=HOUR_IN_SEC * 48):
-            node.run_nodetool(sub_cmd="repair", publish_event=publish_event)
+    def repair_nodetool_repair(self, node=None, publish_event=True, all_nodes=False):
+        if all_nodes:
+            for node in self.cluster.data_nodes:
+                with adaptive_timeout(Operations.REPAIR, node, timeout=HOUR_IN_SEC * 48):
+                    node.run_nodetool(sub_cmd="repair", publish_event=publish_event, ignore_status=True)
+        else:
+            node = node if node else self.target_node
+            with adaptive_timeout(Operations.REPAIR, node, timeout=HOUR_IN_SEC * 48):
+                node.run_nodetool(sub_cmd="repair", publish_event=publish_event)
```

Comment:
lets change this logic. lets try to do without if statement and additional flag
let use something like this:
```python
def repair_nodetool_repair(self, nodes: list | none, timeout: int = HOUR_IN_SEC * 48):
   nodes_to_repair = nodes or [self.target_node]
   for node in nodes_to_repair:
      try:
          with adaptive_timeout(Operations.REPAIR, node, timeout=HOUR_IN_SEC * 48):
                  node.run_nodetool(sub_cmd="repair", publish_event=publish_event)
      except Exception as exc:
               log.warning("repair failed with %s", exc)
```
and need to check each call of repair_nodetool_repair method to update accordingly